### DISCUSSION TITLE
Some problems of data provider width REST API

### DISCUSSION THREAD

**Discussion by zhengtulymGh** [Budibase/budibase#5279] [Help]
rest api:
<img width="1976" alt="v消息20220407-111425" src="https://user-images.githubusercontent.com/13768025/162113077-da815dad-b58c-4596-af9c-10f8d51278cc.png">

Data provider of table width REST API  'getReport'：
<img width="2539" alt="v消息20220407-105147" src="https://user-images.githubusercontent.com/13768025/162111239-d4e0baa8-b653-4f03-b066-26da392eb8c7.png">
<img width="2549" alt="v消息20220407-105147(1)" src="https://user-images.githubusercontent.com/13768025/162111243-d3fb3242-bf02-4e1b-a69a-ad2954c2cb98.png">

Data provider  of option picker width REST API  'getMember':
<img width="2535" alt="v消息20220407-111558" src="https://user-images.githubusercontent.com/13768025/162113649-eb2e4932-783c-439e-b800-1f86f25c1c9f.png">
<img width="2539" alt="v消息20220407-105147(2)" src="https://user-images.githubusercontent.com/13768025/162111248-20a6fd9d-e986-4c19-ad07-dd7a9c83e82b.png">

Query button:
<img width="2545" alt="v消息20220407-105147(3)" src="https://user-images.githubusercontent.com/13768025/162111250-2e34544c-f6cb-4de8-afd6-5af341c3744f.png">

Table:
<img width="2550" alt="v消息20220407-105147(4)" src="https://user-images.githubusercontent.com/13768025/162111253-3af20874-1efe-4557-9379-9e2e1d1add69.png">

The whole contain area refresh when I click 'query' button to update state, width two request being send
<img width="1996" alt="v消息20220407-110307" src="https://user-images.githubusercontent.com/13768025/162111987-c713904d-20f6-47ac-8c12-4cd3d9950f66.png">
<img width="2383" alt="v消息20220407-105030" src="https://user-images.githubusercontent.com/13768025/162111224-840489b9-6a64-4c7e-86ed-982d862bdce2.png">

I would like to be able to  to only refresh the table area.

**Comment by melohagan** [comment]
Hi @zhengtulymGh 

I think both are being refreshed because the option picker is within the top-level data provider, which references the `memberId` state that is updated by the query button. 

You could try moving the option picker outside of the top-level data provider.

**Reply by zhengtulymGh** [reply]
Thanks，moving the option picker outside of the top-level data provider works for me。
